docs/sql/general/constraints.rst

```diff
@@ -53,6 +53,104 @@ constraint or a table constraint.
 For further details about the meaning of the options see
 :ref:`indices_and_fulltext`.
 
+.. _check_constraint:
+
+``CHECK``
+---------
+
+The CHECK constraint specifies that the value/s of certain column/s must satisfy
+a boolean expression on insert and update.
+
+Syntax:
+
+::
+
+  [CONSTRAINT <name>] CHECK (boolean_expression)
+
+If `CONSTAINT <name>` is omitted, CrateDB generates a unique name automatically.
+This name is visible in :ref:`information_schema` and can be used to
+:ref:`ALTER TABLE DROP` a constraint.
+
+::
+
+  cr> select table_schema, table_name, constraint_type, constraint_name
+  ...     from information_schema.table_constraints
+  ...     where table_name = 'metrics1';
+
+The CONSTRAINT definition can either be inline with a column, like this::
+
+    cr> CREATE TABLE metrics1 (
+    ...     weight REAL CONSTRAINT weight_is_positive CHECK (weight >= 0),
+    ... );
+
+Or, also inline, but without explicit name::
+
+    cr> CREATE TABLE metrics2 (
+    ...     weight REAL CHECK (weight >= 0),
+    ... );
+
+Or, on a table level with explicit name:
+
+    cr> CREATE TABLE metrics3 (
+    ...     weight REAL,
+    ...     CONSTRAINT weight_is_positive CHECK (weight >= 0),
+    ... );
+
+Or without name:
+
+    cr> CREATE TABLE metrics4 (
+    ...     weight REAL,
+    ...     CHECK (weight >= 0),
+    ... );
+
+Multiple columns can be referenced:
+
+    cr> CREATE TABLE metrics5 (
+    ...     weight REAL,
+    ...     qty INTEGER,
+    ...     CHECK (weight * qty != 1918),
+    ... );
+
+    cr> CREATE TABLE metrics6 (
+    ...     weight REAL,
+    ...     qty INTEGER CHECK (qty * weight != 1918)
+    ... );
+
+    cr> CREATE TABLE metrics7 (
+    ...     weight REAL,
+    ...     qty INTEGER,
+    ...     CHECK (qty * weight != 1918)
+    ... );
```

Comment:
```suggestion
```

I think one example is enough to illustrate that point. I don't think we need to show all permutations.